### DISCUSSION TITLE
Option to automatically add check-mark on new day.

### DISCUSSION THREAD

**Discussion by ttot1** [iSoron/uhabits#1995] [Feature Requests]
There should be an option to enable "automatic check-marking on the new/current date".

**Description** 
You add new habit or edit an old one and you enable the option, now what happens is that every day at midnight (when the new day starts) the app automatically puts the check-mark on the current/new day.

**Why is it useful?**
This is useful if the habit is based on NOT doing something.
E.G: You want to quit smoking, you create the "No smoking" habit and you enable "automatic check-marking on new day".
Now on the current day when you open the app and look at the check-mark on the current day, it can be interpreted as "you haven't smoked today, you're doing good", but if you happen to relapse on that same day or some other day in the future, the act of the relapse can serve as an "in the moment" reminder to open the app and remove the check-mark from the current day.
This can further encourage the user to NOT do something (like smoking); While they're not smoking, they're saving themselves the trouble of having to remember if they smoked or not the previous day, open the app and add the check-mark for the previous day.

Remembering whether you did or didn't do something the previous day might not seem like an issue for most people, but in any case I believe this can serve to make the tracking and recording of these kind of habits more effortless as long as you keep up the habit of NOT doing something.

Also, I know this might sound similar to [this](https://github.com/iSoron/uhabits/discussions/96) proposal but mine is different in that the app would be adding the check-mark for you at midnight (the new/current day), so it at least saves you the extra step of having to manually add the check-mark on the new day.

Let me know if you guys have any questions or if you guys need me to clarify what I'm describing.

**Comment by essare-rimaz** [comment]
I´d like to add a different perspective to why this could be useful to a broad audience such as folks suffering from ADHD.

As a person who struggles to do things regularly everyday (and who will probably never be able to fully change that) my biggest weakness often isnt the fact that I failed to do my habit, but that with many habits to track, I inevitably fail to in fact track them.

My ideal would be "checkmark by default" not for tracking negative habits or un-habits, but for all my regular habits! Then whenever I fail I click widget to get a fail instead of checkmark.

**Reply by KyleSCraig** [reply]
I don't have ADHD but this sounds like a really bad idea for a few reasons.

- If you forget to untick it when you should, your score rewards it, so it's not as trustworthy of a value. At least the other way around you can say well maybe I did it some of these days but I didn't log it so that's a negative thing resulting in a lower score.
- if you use ? state and hide completed tasks then you can reduce the quantity of habits on the screen. Even if you cross them they won't appear and deciding what to do next is less overwhelming. 
- It's more important to do the habit than it is to mark off the habit, so if you do it but don't log it that's the main part of the goal done and you should be happy about that. Logging is secondary.

**Comment by herrbrixter** [comment]
Duplicate of #125 

Related to #696
